src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> t = tail;
+        Node<T> newNode = new Node<>(t, value, null);
+        tail = newNode;
+        if (t == null) {
+            head = newNode;
+        } else {
+            t.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nextNode = isIndexInTheList(index);
+            Node<T> prevNode = nextNode.prev;
+            Node<T> newNode = new Node<>(prevNode, value, nextNode);
+            nextNode.prev = newNode;
+            if (prevNode == null) {
+                head = newNode;
+            } else {
+                prevNode.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        final int sizeOfInputList = list.toArray().length;
+        checkPositionIndex(size);
+        Node<T> prev;
+        Node<T> next;
+        if (size == size) {
+            next = null;
+            prev = tail;
+        } else {
+            next = isIndexInTheList(size);
+            prev = next.prev;
+        }
+        for (Object elementOfInputList : list.toArray()) {
+            T element = (T) elementOfInputList;
+            Node<T> newNode = new Node<>(prev, element, null);
+            if (prev == null) {
+                head = newNode;
+            } else {
+                prev.next = newNode;
+            }
+            prev = newNode;
+        }
+        if (next == null) {
+            tail = prev;
+        } else {
+            prev.next = next;
+            next.prev = prev;
+        }
+        size += sizeOfInputList;
```

Comment:
This solution is too difficult. You can use your add(index) method
```suggestion
       for (T t : list) {
            add(t);
        }
```